src/transaction/log_recovery_redo_parallel.cpp

```diff
@@ -673,10 +704,41 @@ namespace cublog
 
     for (unsigned task_idx = 0; task_idx < a_task_count; ++task_idx)
       {
-	// NOTE: task ownership goes to the worker pool
-	auto task = new redo_task (task_idx, m_task_state_bookkeeping, m_job_queue);
-	m_worker_pool->execute (task);
+	auto task = std::make_unique<redo_parallel::redo_task> (task_idx, m_task_state_bookkeeping, m_job_queue);
+	m_worker_pool->execute (task.get ());
+	m_redo_tasks.push_back (std::move (task));
+      }
+  }
+
+  void
+  redo_parallel::log_perf_stats () const
+  {
+    const cubperf::statset_definition definition
+    {
+      perf_stats_async_definition_init_list
+    };
+
+    std::vector<cubperf::stat_value> accum_perf_stat_results;
+    accum_perf_stat_results.resize (definition.get_value_count ());
+
+    for (auto &redo_task: m_redo_tasks)
+      {
+	redo_task->accumulate_perf_stats (accum_perf_stat_results.data (), accum_perf_stat_results.size ());
+	redo_task->log_perf_stats ();
       }
+
+    // average
+    const std::size_t task_count = m_redo_tasks.size ();
+    const std::size_t value_count = accum_perf_stat_results.size ();
+    std::vector<double> avg_perf_stat_results;
```

Comment:
Having double is not really necessary, cubperf::stat_value should be good enough. As a result `log_perf_stats_values_with_definition()` may become a simple function, without template.
```suggestion
    std::vector<cubperf::stat_value> avg_perf_stat_results;
```